### PR TITLE
Allow popup to route correct coordinates to Parent.

### PR DESCRIPTION
**Pull request by danwalmsley**
When a user clicks inside a popup that doesnt `e.Handled = true` the pointer event, the event is also raised on the Popups parent.

Which has a different "VisualRoot". (Window vs PopupRoot)

when the user handles the event on the control inside the parent window, and calls `e.GetPosition(this)` (passing any visual that isnt a child of the popup), they currently always get 0,0.

why? because the `_rootVisual` of `PointerEventArgs` (which originated on the popup) is different to the root of the visual passed to GetPosition.

How was this fixed?

If GetPosition sees that the _rootVisal and the relativeTo.VisualRoot are different, it converts to absolute screen coordinates using the eventargs _rootVisual, and then converts from Screen coordinates to client coordinates of the visual passed to GetPosition.

It has been determined by @kekekeks that this is not an issue and can be implemented on all platforms. Since Wayland now exposes:
https://cgit.freedesktop.org/wayland/wayland-protocols/tree/stable/xdg-shell/xdg-shell.xml#n1276

### CODE REVIEW

**Review by maxkatz6 — APPROVED** [review]

**Comment by maxkatz6** [comment]
Seems like we can have this even with wayland now, so should be fine.

File: src/Avalonia.Base/Input/PointerEventArgs.cs [hunk 1]
```diff
@@ -70,6 +70,13 @@ private Point GetPosition(Point pt, Visual? relativeTo)
             if (relativeTo == null)
                 return pt;
 
+            if (!ReferenceEquals(_rootVisual, relativeTo.VisualRoot) && relativeTo.VisualRoot is Visual v)
```
Comment (grokys) [inline src/Avalonia.Base/Input/PointerEventArgs.cs:73]: Could we get a comment here to explain what it's doing/why it's needed?

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035340-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.Base/Input/PointerEventArgs.cs (hunk 1 above)
Comment (danwalmsley) [reply src/Avalonia.Base/Input/PointerEventArgs.cs]: added comments.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035417-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0035475-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**